### ISSUE TITLE
Docker build issue

### ISSUE THREAD

**Issue by AnanthGopal** [strongloop/loopback-next#2987]
When i try to release loopback application in docker that time i got below error 

`added 519 packages from 1472 contributors and audited 4909 packages in 21.532s
found 3 vulnerabilities (1 low, 2 moderate)
  run `npm audit fix` to fix them, or `npm audit` for details
/loobpack # npm start --rebuild

> new@1.0.0 prestart /loobpack
> npm run build

> new@1.0.0 build /loobpack
> lb-tsc es2017 --outDir dist

src/controllers/user.controller.ts:47:10 - error TS2339: Property 'password' does not exist on type 'UserWithoutId'.

47     user.password = await this.passwordHahser.hashPassword(user.password);
            ~~~~~~~~

src/controllers/user.controller.ts:47:65 - error TS2339: Property 'password' does not exist on type 'UserWithoutId'.

47     user.password = await this.passwordHahser.hashPassword(user.password);
                                                                   ~~~~~~~~

src/controllers/user.controller.ts:49:22 - error TS2339: Property 'password' does not exist on type 'User'.

49     delete savedUser.password;
                        ~~~~~~~~

src/controllers/validation/user.validation.ts:32:66 - error TS2345: Argument of type '{ where: { userName: { ilike: string; }; }; }' is not assignable to parameter of type 'Filter<User>'.
  Types of property 'where' are incompatible.
    Type '{ userName: { ilike: string; }; }' is not assignable to type 'Condition<User> | AndClause<User> | OrClause<User> | undefined'.
      Object literal may only specify known properties, and 'userName' does not exist in type 'Condition<User> | AndClause<User> | OrClause<User>'.

32         let userDetail = await userRepository.findOne({ where: { userName: { ilike: userName } } });
                                                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/controllers/validation/user.validation.ts:56:31 - error TS2339: Property 'userName' does not exist on type 'UserWithoutId'.

56     if (!isValidEmail(request.userName)) {
                                 ~~~~~~~~

src/controllers/validation/user.validation.ts:60:17 - error TS2339: Property 'password' does not exist on type 'UserWithoutId'.

60     if (request.password.length < 8) {
                   ~~~~~~~~

src/framework/services/JWT.authentication.service.ts:31:16 - error TS2345: Argument of type '{ where: { userName: { ilike: string; }; }; }' is not assignable to parameter of type 'Filter<User>'.
  Types of property 'where' are incompatible.
    Type '{ userName: { ilike: string; }; }' is not assignable to type 'Condition<User> | AndClause<User> | OrClause<User> | undefined'.
      Object literal may only specify known properties, and 'userName' does not exist in type 'Condition<User> | AndClause<User> | OrClause<User>'.

31       where: { userName: { ilike: request.userName } },
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/models/userwithoutid.model.ts:2:34 - error TS2307: Cannot find module './userLoginRequest.model'.

2 import { UserLoginRequest } from './userLoginRequest.model';
                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~

src/repositories/user.repository.ts:13:3 - error TS2344: Type 'User' does not satisfy the constraint 'Entity'.
  Type 'User' is missing the following properties from type 'Entity': getId, getIdObject, toJSON, toObject

13   User,
     ~~~~

src/repositories/user.repository.ts:20:11 - error TS2345: Argument of type 'typeof User' is not assignable to parameter of type 'typeof Entity & { prototype: User; }'.
  Type 'typeof User' is missing the following properties from type 'typeof Entity': getIdOf, buildWhereForId, modelName, definition

20     super(User, dataSource);
             ~~~~


Found 10 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! new@1.0.0 build: `lb-tsc es2017 --outDir dist`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the new@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-05-29T13_14_02_208Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! new@1.0.0 prestart: `npm run build`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the new@1.0.0 prestart script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-05-29T13_14_02_308Z-debug.log`


I try to resolve this error i can't

**Comment by raymondfeng** [comment]
How did you create the docker image? Can you get versions of node, typescript, and others?

```sh
node -v
npm ls
```

**Comment by secombrian** [comment]
I've got the same problem when updated to the latest version via NPM.
Node: v8.11.3
+-- @loopback/authentication@1.0.17
| +-- @loopback/context@1.8.1 deduped
| +-- @loopback/core@1.2.1 deduped
| +-- @loopback/metadata@1.1.7
| +-- @loopback/openapi-v3@1.3.2 deduped
| +-- @loopback/rest@1.9.1 deduped
| +-- passport@0.4.0 deduped
| `-- passport-strategy@1.0.0
+-- @loopback/boot@1.1.2
| +-- @loopback/context@1.8.1 deduped
| +-- @loopback/core@1.2.1 deduped
| +-- @loopback/repository@1.2.1 deduped
| +-- @loopback/service-proxy@1.1.1 deduped
| +-- @types/debug@4.1.4
| +-- @types/glob@7.1.1
| +-- debug@4.1.1
| `-- glob@7.1.4
+-- @loopback/build@1.4.0
| +-- @loopback/tslint-config@2.0.4
| +-- @types/mocha@5.2.7
| +-- @types/node@10.14.8 deduped
| +-- cross-spawn@6.0.5
| +-- debug@4.1.1 deduped
| +-- fs-extra@7.0.1
| +-- glob@7.1.4 deduped
| +-- mocha@6.1.4
| +-- nyc@13.3.0
| +-- prettier@1.17.1
| +-- rimraf@2.6.3
| +-- source-map-support@0.5.12
| +-- strong-docs@4.2.0
| +-- tslint@5.17.0
| `-- typescript@3.5.1
+-- @loopback/context@1.8.1
| +-- @loopback/metadata@1.1.7 deduped
| +-- debug@4.1.1 deduped
| +-- p-event@4.1.0
| `-- uuid@3.3.2
+-- @loopback/core@1.2.1
| `-- @loopback/context@1.8.1 deduped
+-- @loopback/openapi-v3@1.3.2
| +-- @loopback/context@1.8.1 deduped
| +-- @loopback/openapi-v3-types@1.0.21
| +-- @loopback/repository-json-schema@1.5.1
| +-- debug@4.1.1 deduped
| `-- lodash@4.17.11 deduped
+-- @loopback/repository@1.2.1
| +-- @loopback/context@1.8.1 deduped
| +-- @loopback/core@1.2.1 deduped
| +-- @types/debug@4.1.4 deduped
| +-- debug@4.1.1 deduped
| +-- lodash@4.17.11 deduped
| `-- loopback-datasource-juggler@4.7.1
+-- @loopback/rest@1.9.1
| +-- @loopback/context@1.8.1 deduped
| +-- @loopback/core@1.2.1 deduped
| +-- @loopback/http-server@1.2.6
| +-- @loopback/openapi-v3@1.3.2 deduped
| +-- @loopback/openapi-v3-types@1.0.21 deduped
| +-- @types/body-parser@1.17.0
| +-- @types/cors@2.8.5
| +-- @types/express@4.16.1
| +-- @types/express-serve-static-core@4.16.6
| +-- @types/http-errors@1.6.1
| +-- @types/on-finished@2.3.1
| +-- @types/serve-static@1.13.2
| +-- @types/type-is@1.6.2
| +-- ajv@6.10.0
| +-- body-parser@1.19.0
| +-- cors@2.8.5
| +-- debug@4.1.1 deduped
| +-- express@4.17.1
| +-- http-errors@1.7.2
| +-- js-yaml@3.13.1
| +-- lodash@4.17.11 deduped
| +-- on-finished@2.3.0
| +-- openapi-schema-to-json-schema@2.2.0
| +-- openapi3-ts@1.3.0
| +-- path-to-regexp@3.0.0
| +-- qs@6.7.0
| +-- strong-error-handler@3.2.0
| +-- type-is@1.6.18
| `-- validator@10.11.0
+-- @loopback/rest-explorer@1.1.13
| +-- @loopback/context@1.8.1 deduped
| +-- @loopback/core@1.2.1 deduped
| +-- @loopback/rest@1.9.1 deduped
| +-- ejs@2.6.1
| `-- swagger-ui-dist@3.22.2
+-- @loopback/service-proxy@1.1.1
| +-- @loopback/context@1.8.1 deduped
| +-- @loopback/core@1.2.1 deduped
| `-- loopback-datasource-juggler@4.7.1 deduped
+-- @loopback/testlab@1.2.1
| +-- @types/express@4.16.1 deduped
| +-- @types/fs-extra@5.1.0
| +-- @types/shot@4.0.0
| +-- @types/sinon@7.0.12
| +-- @types/supertest@2.0.7
| +-- express@4.17.1 deduped
| +-- fs-extra@7.0.1 deduped
| +-- oas-validator@3.3.0
| +-- shot@4.0.7
| +-- should@13.2.3
| +-- sinon@7.3.2
| `-- supertest@4.0.2
+-- @loopback/tslint-config@1.0.0
+-- @types/bcryptjs@2.4.2
+-- @types/moment-timezone@0.5.12
| `-- moment@2.24.0 deduped
+-- @types/node@10.14.8
+-- @types/passport@1.0.0
| `-- @types/express@4.16.1 deduped
+-- @types/passport-http@0.3.8
| +-- @types/express@4.16.1 deduped
| `-- @types/passport@1.0.0 deduped
+-- @types/uniqid@4.1.3 extraneous
+-- bcryptjs@2.4.3
+-- jsonwebtoken@8.5.1
| +-- jws@3.2.2
| +-- lodash.includes@4.3.0
| +-- lodash.isboolean@3.0.3
| +-- lodash.isinteger@4.0.4
| +-- lodash.isnumber@3.0.3
| +-- lodash.isplainobject@4.0.6
| +-- lodash.isstring@4.0.1
| +-- lodash.once@4.1.1
| +-- ms@2.1.1
| `-- semver@5.7.0
+-- lodash@4.17.11
+-- loopback-connector-mongodb@4.0.0
| +-- async@2.6.2
| +-- bson@1.1.1
| +-- debug@3.2.6
| +-- loopback-connector@4.7.0
| +-- mongodb@3.2.6
| `-- strong-globalize@4.1.3
+-- memory@0.0.3
+-- moment@2.24.0
+-- moment-timezone@0.5.25
| `-- moment@2.24.0 deduped
+-- passport@0.4.0
| +-- passport-strategy@1.0.0 deduped
| `-- pause@0.0.1
+-- passport-http@0.3.0
| `-- passport-strategy@1.0.0 deduped

**Comment by raymondfeng** [comment]
The latest version of `@loopback/repository` is v1.6.0. We fixed the compilation issue with TS 3.5 recently. Please upgrade to latest versions with `npm up`.

**Comment by dhmlau** [comment]
Closing as resolved. See [above comment ](https://github.com/strongloop/loopback-next/issues/2987#issuecomment-498519851). @AnanthGopal @secombrian, if you're encountering a similar issue, please feel free to open a new issue. Thanks.
